bedrock/src/secure/mod.rs

```diff
@@ -0,0 +1,356 @@
+use bedrock_macros::{bedrock_error, bedrock_export};
+use dryoc::kdf::Kdf;
+use rand::{rngs::OsRng, RngCore};
+use ruint::aliases::U256;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use sha2::{Digest, Sha256};
+
+const CONTEXT: [u8; 8] = *b"OXIDEKEY";
+const ETHEREUM_KEY_ID: u64 = 0x00;
+const ROTATED_ETHEREUM_KEY_ID: u64 = 0x01;
+const WORLDID_KEY_ID: u64 = 0x10;
+//const ORB_ENCRYPTION_KEY_ID: u64 = 0x20;
+const MARBLE_SEED_KEY_ID: u64 = 0x30;
+//const DOCUMENT_PCP_KEY_ID: u64 = 0x40;
+const WORLDCHAT_BACKUP_KEY_ID: u64 = 0x50;
+const WORLD_CHAT_PUSH_ID_KEY_ID: u64 = 0x60;
+const KEY_LENGTH: usize = 32;
+const ROTATED_ETHEREUM_KEY_SALT: &str = "ethereum_key_rotated_";
+const MARBLE_SEED_SALT: &str = "world_id_card_marble_b9dcc41bf41_";
+
+type Key = [u8; KEY_LENGTH];
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
```

Comment:
Do we need Debug here? In other projects we overwrite the impl such that it only prints it redacted if used.